server/routes/creatingLicences/handlers/additionalLicenceConditionUploadsHandler.ts

```diff
@@ -0,0 +1,57 @@
+import { Request, Response } from 'express'
+import LicenceService from '../../../services/licenceService'
+import {
+  currentOrNextSequenceForCondition,
+  getAdditionalConditionByCode,
+  getAdditionalConditionType,
+} from '../../../utils/conditionsProvider'
+import { AddAdditionalConditionRequest } from '../../../@types/licenceApiClientTypes'
+import conditionType from '../../../enumeration/conditionType'
+
+export default class AdditionalLicenceConditionUploadsHandler {
+  constructor(private readonly licenceService: LicenceService) {}
+
+  GET = async (req: Request, res: Response): Promise<void> => {
+    const { licenceId, conditionCode } = req.params
+    const { user } = res.locals
+    const licence = await this.licenceService.getLicence(licenceId, user)
+
+    const conditions = licence.additionalLicenceConditions.filter(c => c.code === conditionCode)
+
+    if (conditions.length === 0) {
+      return res.redirect(`/licence/create/id/${licence.id}/additional-licence-conditions/callback?fromReview=true`)
+    }
+
+    return res.render('pages/create/fileUploads', { conditions, licenceId, conditionType })
+  }
+
+  POST = async (req: Request, res: Response): Promise<void> => {
+    const { licenceId } = req.params
+    const { user, licence } = res.locals
+    const { uploadFile, conditionCode } = req.body
+
+    const condition = getAdditionalConditionByCode(conditionCode)
+    const type = getAdditionalConditionType(conditionCode)
+
+    if (uploadFile !== 'yes') {
+      return res.redirect(`/licence/create/id/${licenceId}/check-your-answers`)
+    }
+
+    const sequence = currentOrNextSequenceForCondition(licence.additionalLicenceConditions, conditionCode)
```

Comment:
I'd tend to shove stuff like this and the calls to things like  the `getAdditionalConditionByCode`calls into the service.
Bit of a personal preference around having very thin controllers which are solely responsible with binding to requests and creating responses.